src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,284 @@
+            regPool = pool;
+        }
+
+    private:
+        TReg reg;
+        RegistersPool::Ptr regPool;
+    };
+
+    virtual ~RegistersPool() {
+        checkUniqueAndUpdate(false);
+    }
+
+    template <x64::cpu_isa_t isa>
+    static Ptr create(std::initializer_list<Xbyak::Reg> regsToExclude);
+
+    template<typename TReg>
+    size_t countFree() const {
+        if (std::is_same<TReg, Xbyak::Xmm>::value || std::is_same<TReg, Xbyak::Ymm>::value || std::is_same<TReg, Xbyak::Zmm>::value) {
+            return simdSet.countUnused();
+        } else if (std::is_same<TReg, Xbyak::Reg64>::value || std::is_same<TReg, Xbyak::Reg32>::value) {
+            return generalSet.countUnused();
+        } else if (std::is_same<TReg, Xbyak::Opmask>::value) {
+            return countUnusedOpmask();
+        }
+    }
+
+protected:
+    class PhysicalSet {
+    public:
+        PhysicalSet(int size) : isFreeIndexVector(size, true) {}
+
+        void setAsUsed(int regIdx) {
+            if (regIdx >= isFreeIndexVector.size() || regIdx < 0) {
+                IE_THROW() << "regIdx is out of bounds in RegistersPool::PhysicalSet::setAsUsed()";
+            }
+            if (!isFreeIndexVector[regIdx]) {
+                IE_THROW() << "Inconsistency in RegistersPool::PhysicalSet::setAsUsed()";
+            }
+            isFreeIndexVector[regIdx] = false;
+        }
+
+        void setAsUnused(int regIdx) {
+            if (regIdx >= isFreeIndexVector.size() || regIdx < 0) {
+                IE_THROW() << "regIdx is out of bounds in RegistersPool::PhysicalSet::setAsUsed()";
+            }
+            if (isFreeIndexVector[regIdx]) {
+                IE_THROW() << "Inconsistency in RegistersPool::PhysicalSet::setAsUnused()";
+            }
+            isFreeIndexVector[regIdx] = true;
+        }
+
+        int getUnused(int requestedIdx) {
+            if (requestedIdx == anyIdx) {
+                return getFirstFreeIndex();
+            } else {
+                if (!isFreeIndexVector[requestedIdx]) {
+                    IE_THROW() << "The register with index #" << requestedIdx << " already used in the RegistersPool";
+                }
+                return requestedIdx;
+            }
+        }
+
+        void exclude(Xbyak::Reg reg) {
+            isFreeIndexVector.at(reg.getIdx()) = false;
+        }
+
+        size_t countUnused() const {
+            size_t count = 0;
+            for (const auto& isFree : isFreeIndexVector) {
+                if (isFree) {
+                    ++count;
+                }
+            }
+            return count;
+        }
+
+    private:
+        int getFirstFreeIndex() {
+            for (int c = 0; c < isFreeIndexVector.size(); ++c) {
+                if (isFreeIndexVector[c]) {
+                    return c;
+                }
+            }
+            IE_THROW() << "Not enough registers in the RegistersPool";
+        }
+
+    private:
+        std::vector<bool> isFreeIndexVector;
```

Comment:
It looks like the std::set of unused indices would be more useful here. It will simplify all of these functions.

Reply:
It previously was implemented using `std::unordered_set`, but using set it is hard to implement allocation register with specific `requestedIdx`.
`isFreeIndexVector` simplify and makes predictable requesting from RegistersPool concrete idx

Reply:
ok